pkg/scheduler/framework/plugins/nodeports/node_ports.go

```diff
@@ -61,10 +61,10 @@ func (pl *NodePorts) Name() string {
 func getContainerPorts(pods ...*v1.Pod) []*v1.ContainerPort {
 	ports := []*v1.ContainerPort{}
 	for _, pod := range pods {
-		for j := range pod.Spec.Containers {
-			container := &pod.Spec.Containers[j]
-			for k := range container.Ports {
-				ports = append(ports, &container.Ports[k])
+		for _, container := range pod.Spec.Containers {
```

Comment:
I think this is a duplicate of https://github.com/kubernetes/kubernetes/pull/93745 which has already received several feedbacks.

Reply:
Thanks for your feedback! I have tried to take the point about not wanting to create _new_ references into consideration. Not sure if this is any more readable.

Reply:
Not worth introducing this change IMO: https://github.com/kubernetes/kubernetes/pull/93823#discussion_r468085964